lib/DelegatedModuleFactoryPlugin.js

```diff
@@ -35,8 +36,18 @@ class DelegatedModuleFactoryPlugin {
 					const dependency = data.dependencies[0];
 					const request = dependency.request;
 					if (request && request.indexOf(scope + "/") === 0) {
-						const innerRequest =
-							"." + request.substr(scope.length) + this.options.mainFiles[0];
+						const innerRequest = "." + request.substr(scope.length);
+						for (let i = 0; i < this.options.mainFiles.length; i++) {
```

Comment:
I made some kind of cartesian product using the extensions and the mainFile so in that way I'm making sure to try without the mainFiles and for each file name on the array.